sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -150,6 +153,14 @@ class DataFrame private[sql](
     }
   }
 
+  private[sql] def resolveToIndex(colName: String): Option[Int] = {
+    val resolver = sqlContext.analyzer.resolver
+    // First remove any user supplied quotes.
+    val unquotedColName = colName.stripPrefix("`").stripSuffix("`")
```

Comment:
for example, what if a column is named ``` `a`a` ```? User should be able to just pass in ``` `a`a` ``` and we shouldn't strip the "`"